game.rb

```diff
@@ -0,0 +1,118 @@
+
+# game.rb
+require_relative 'board'
+require_relative 'player'
+require_relative 'computer'
+
+# Deals with the running of the game and the rules
+class Game
+  attr_accessor :codemaker, :codebreaker
+  
+  $colors = %w(R O Y G B P)
+
+  def initialize
+    @board = Board.new
+  end
+
+  def initial_setup
+    puts "Do you want to be the codemaker or codebreaker?"
+    puts "Enter M or B"
+    puts "**********************************************"
+
+    answer = gets.chomp.strip
+    if(answer.upcase == "M")
+      @codemaker = Player.new
+      @codebreaker = Computer.new
+    elsif(answer.upcase == "B")
+      @codemaker = Computer.new
+      @codebreaker = Player.new
+    else 
+      puts "Please ensure you enter your input correctly"
+      initial_setup
+    end
+  end
+
+  # The game starts here
+  def play
+    
+    puts "Welcome to Mastermind"
+    puts "**********************************************"
+
+    initial_setup
+    code = @codemaker.generate_code
+    12.times do |i| 
+      puts "This is guess number #{i+1}"
+      guess = @codebreaker.get_guess
+      @board.add_guess(guess)
+      feedback = give_feedback(code, guess)
+
+      @board.render(feedback)
+
```

Comment:
Lines 51-56 should also be refactored according to `winner?` method (see below).
```ruby
  success_message if winner?
  reveal_code if i == 11       # reveal code can be renamed to game_over
```
Exit the game with a goodbye message in a `goodbye` method and call `goodbye` at the end of `success_message` and `reveal_code`.